### DISCUSSION TITLE
2.4-preview

### DISCUSSION THREAD

**Discussion by arianneorpilla** [arianneorpilla/jidoujisho#158] [Announcements]
This preview release implements several fixes and a dictionary schema overhaul. Note that only Japanese language and the Yomichan Dictionary format is available for this preview. This release will require a fresh installation. 

**System Requirements**: This application has been tested to work on a variety of devices and emulators running **Android 7 to 13**. Please note that a significant amount of internal storage space may be taken up by the dictionary database for indexing in order to  optimize query speed and performance. Try to give the application around **2GB of storage space**.

# 2.4-preview

### 📖 Dictionary Improvements
- **Dictionary System Overhaul** - The internal dictionary schema has been overhauled, with significant improvements to search and import speed, as well as sorting of dictionary search results.
- **Wildcards Search** - Searching with wildcards has now been implemented. `?` or `？` will now match exactly a single character, while `*` or `※`  (easily auto-suggested as `こめ`) will match zero or any number of characters.
- Fixed an issue where some frequency dictionaries would not import correctly. (**#157**)
- Single dictionaries can no longer be deleted one by one, instead an option to clear data in the dictionary database has been added.  

### 🛠 Miscellaneous Changes
- Using the app as an external player for Jellyfin will now pass the selected subtitle of played videos into the application. (**#91**)
- Fixed an issue where the sentence field would be blank if a played video's media source does not match the currently selected player media source in the home tab.
- Changes to the cache mode of the Reader WebView may prevent data expiration. (**#147**)
- Fixed an issue related to poly-fills when importing a dictionary with a certain (**#153**)
- Incognito mode has been removed. 

# 📲 Which file should I download?
The files below are for different device architectures. As a rule of thumb, `arm64` should work for most modern mobile devices. `armeabi` should work on older devices. Generally, desktops and emulators should use `x86_64`.

Depending on which device you are using, download the appropriate `APK` by expanding the `Assets` drop-down below and install on your mobile phone by launching the `APK` with your file explorer.

<hr /><em>This discussion was created from the release <a href='https://github.com/lrorpilla/jidoujisho/releases/tag/2.4-preview'>2.4-preview</a>.</em>

**Comment by TheAtlasRises** [comment]
It seems like in the latest preview, the pitch accent dictionary is broken. Edit: this is the Kanjium Pitch Accent Dictionary. The device is the Galaxy A52s on Android 13.

With the pitch accent dictionary set as "hide":
![Screenshot_20230306_101007](https://user-images.githubusercontent.com/106642434/223119429-bc917783-6192-4313-8b7d-af2cf504f3b8.jpg)

With the pitch accent dictionary set as "show":
![Screenshot_20230306_101033](https://user-images.githubusercontent.com/106642434/223119580-4b47c7de-84e5-4860-9eb3-05da228caaa0.jpg)

**Reply by arianneorpilla** [reply]
Hotfixed

**Reply by arianneorpilla** [reply]
Thank you for the report

**Reply by arianneorpilla** [reply]
Reuploaded just now since I missed something
